tests/utils/sampler_test.py

```diff
@@ -1,36 +1,35 @@
-from pathlib import Path
-
 import numpy as np
 import pytest
+import torch
 from scipy.stats import chisquare
 
-from fl4health.utils.dataset import MnistDataset
+from fl4health.utils.dataset import SyntheticDataset
 from fl4health.utils.random import set_all_random_seeds, unset_all_random_seeds
 from fl4health.utils.sampler import DirichletLabelBasedSampler, MinorityLabelBasedSampler
 
 
+def construct_synthetic_dataset() -> SyntheticDataset:
+    # set seed for creation
+    torch.manual_seed(42)
+    random_inputs = torch.rand((10000, 3, 3))
+    # Note high is exclusive here
+    random_labels = torch.randint(low=0, high=10, size=(1000, 1))
+    # unset seed thereafter
+    torch.seed()
+    return SyntheticDataset(random_inputs, random_labels)
+
+
+SYNTHETIC_DATASET = construct_synthetic_dataset()
+
+
 def test_minority_sampler() -> None:
     minority_numbers = set([1, 2, 3])
     downsampling_ratio = 0.1
     sampler = MinorityLabelBasedSampler(
         unique_labels=list(range(10)), downsampling_ratio=downsampling_ratio, minority_labels=minority_numbers
     )
 
-    # Training
-    ds = MnistDataset(data_path=Path("examples/datasets/MNIST"), train=True)
-    samples_per_class = [ds.targets[ds.targets == i].size(0) for i in range(10)]
-
-    ds_new = sampler.subsample(ds)
-
-    new_samples_per_class = [
-        int(num_samples * downsampling_ratio) if i in minority_numbers else num_samples
-        for i, num_samples in enumerate(samples_per_class)
-    ]
-
-    assert len(ds_new) == sum(new_samples_per_class)
-
-    # Testing
-    ds = MnistDataset(data_path=Path("examples/datasets/MNIST"), train=False)
+    ds = SYNTHETIC_DATASET
```

Comment:
We were essentially running the same test and the two different MNIST splits, which seemed sort of redundant. So I just did the one.